### PR TITLE
[New Feature] Add the serving support

### PR DESCRIPTION
**Pull request by cyber-pioneer**
## New Feature
Flagscale supports serve.
 ## Description
This pull request introduces support for deploying large models with FlagScale, leveraging the Ray framework for efficient orchestration and scalability. Currently, this implementation supports the Qwen model, enabling users to easily deploy and manage large-scale machine learning services.

Future Key features include:

- Easy distributed Serve on base of eamless integration with Ray.
- Optimized resource management for serve excution of multiple tasks.
- Simplified deployment process for the LLM and Multimodal models.

More details in [Serve](https://github.com/FlagOpen/FlagScale/blob/dd8a68f9120d3851d0b0cad57e3d9523038fac40/flagscale/serve/README.md).

### CODE REVIEW

File: examples/qwen/serve/qwen2.5_7b.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+command-line-mode: true
+
+vllm:
+  model-tag: /models/Qwen2.5-7B-Instruct
+  tensor-parallel-size: 1
+  gpu-memory-utilization: 0.9
+  max-model-len: 32768
+  max-num-seqs: 256
+  port: 4567
```
Comment (ceci3) [inline examples/qwen/serve/qwen2.5_7b.yaml:9]: port也是vllm里需要的参数吗？

File: examples/qwen/serve/qwen2.5_7b.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  action-args:
```
Comment (ceci3) [inline examples/qwen/serve/qwen2.5_7b.yaml:10]: action-args和上面的字段有什么区别？什么字段会放在这里

File: flagscale/runner/runner_serve.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+                    ssh_port,
+                    dryrun,
+                )
+
+            # Step 3: run the host_run_script_file on the remote host
+            run_ssh_command(host, f"bash {host_run_script_file}", ssh_port, dryrun)
+        else:
+            run_local_command(f"bash {host_run_script_file}", dryrun)
+
+    def run(self, with_test=False, dryrun=False):
```
Comment (ceci3) [inline flagscale/runner/runner_serve.py:218]: with_test的含义是要不要放在后台执行吗？

File: flagscale/runner/runner_serve.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+            nproc_per_node = get_nproc_per_node(
+                None, nproc_from_args, num_visible_devices
+            )
+            available_addr = runner_config.get("master_addr", "localhost")
+            available_port = runner_config.get("master_port", get_free_port())
+            self._run_each(
+                "localhost",
+                available_addr,
+                available_port,
+                1,
```
Comment (ceci3) [inline flagscale/runner/runner_serve.py:266]: local的话只支持单节点单卡吗？

File: flagscale/serve/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## Setup
+[Install vLLM](../../../README.md#setup)
+
+[Prepare Qwen data](https://www.modelscope.cn/models/Qwen/Qwen2.5-7B-Instruct/summary)
+
+```shell
+pip install modelscope
+modelscope download --model Qwen/Qwen2.5-7B-Instruct --local_dir /models/
```
Comment (ceci3) [inline flagscale/serve/README.md:23]: 准备模型的流程和上面一样加一个小标题

File: flagscale/serve/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+
+## Serve run
+```shell
+cd FlagScale
+python run.py --config-path examples/qwen/ --config-name config action=run
+```
+
+
```
Comment (ceci3) [inline flagscale/serve/README.md]: 后续可以加一个文档说明怎么替换模型和数据，用户怎么使用工具来部署自己的模型

File: flagscale/serve/run_simple_vllm.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    stdout, stderr = process.communicate()
+
+    logger.info("Standard Output:")
+    logger.info(stdout)
+    # logger.info(stdout.decode())
+    logger.info("Standard Error:")
+    logger.info(stderr)
+    # logger.info(stderr.decode())
+
```
Comment (ceci3) [inline flagscale/serve/run_simple_vllm.py]: 可以精简一下

File: examples/qwen/serve/qwen2.5_7b.yaml (hunk 1 above)
Comment (cyber-pioneer) [reply examples/qwen/serve/qwen2.5_7b.yaml:9]: 是的 vllm字段内的参数，与原生的vllm serve命令行保持一致

File: examples/qwen/serve/qwen2.5_7b.yaml (hunk 2 above)
Comment (cyber-pioneer) [reply examples/qwen/serve/qwen2.5_7b.yaml:10]: 举例说明：vllm serve /models/Qwen2.5-7B-Instruct --tensor-parallel-size=1 --gpu-memory-utilization=0.9 --max-model-len=32768 --max-num-seqs=256 --port=4567 --trust-remote-code --enable-chunked-prefill

--trust-remote-code --enable-chunked-prefill 这两个参数是预设的行为，命令行里是禁止传入参数值的，属于action args。 

本PR将增加文档说明部署参数选项，降低开发者的接入成本。

File: flagscale/runner/runner_serve.py (hunk 4 above)
Comment (cyber-pioneer) [reply flagscale/runner/runner_serve.py:266]: 本pr里仅支持单卡，多卡场景支持中

File: flagscale/serve/README.md (hunk 5 above)
Comment (cyber-pioneer) [reply flagscale/serve/README.md:23]: ok

File: flagscale/serve/README.md (hunk 6 above)
Comment (cyber-pioneer) [reply flagscale/serve/README.md]: 好的，本pr会补充

File: flagscale/serve/run_simple_vllm.py (hunk 7 above)
Comment (cyber-pioneer) [reply flagscale/serve/run_simple_vllm.py]: ok

File: flagscale/runner/runner_serve.py (hunk 3 above)
Comment (cyber-pioneer) [reply flagscale/runner/runner_serve.py:218]: 是的

**Review by aoyulong — APPROVED** [review]
LGTM

**Pull request merged**